phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/container/skin/ContainerEngineSettingsPanelSkin.java

```diff
@@ -85,7 +89,15 @@ private void updateEngineSettingsGrid(final GridPane engineSettingsGrid) {
 
             // if the container is not specified set no default values
             if (container != null) {
-                engineSettingComboBox.setValue(engineSetting.getCurrentOption(container.getName()));
+                try {
+                    engineSettingComboBox.setValue(engineSetting.getCurrentOption(container.getName()));
+                } catch (Exception e) {
```

Comment:
Is it possible to catch something more specific here? I would expect that only scripting exceptions are thrown here right?

Reply:
I don't think so. For example, we can also have `PolyglotException`.

Reply:
I think we should only call certain types of exceptions. Everything else links to a more drastic problem which we can't solve in this catch block

Reply:
I'm not sure. In my opinion, this is exactly the desired behavior: Try to get the currently used option. If that's not possible for whatever reason, fall back to default. I doesn't make sense to affect any other parts of Phoenicis by not catching the exception here.

Reply:
In my opinion there are exceptions that should lead to a software crash. Or at least not to a fallback to a default option, for example security related exceptions.

Reply:
Like e.g.?

Reply:
For example: https://docs.oracle.com/javase/7/docs/api/java/lang/SecurityException.html

Reply:
Even in that case: why should we allow this to crash Phoenicis? I don't see how anything bad can happen if we catch the exception.

Reply:
You are right in that we can (and maybe even should) catch such exceptions as well. The issue is just that we shouldn't simply do a fallback to a default option if such an exception is thrown. At least a popup showing an error message to the user should be opened

Reply:
I'm not sure what we could show in the popup. What do we expect the user to do?

Reply:
Check his repositories if they are trustworthy and/or updates are available. The official repositories should not throw such exceptions.

Reply:
It's not necessarily related to the repository. If you check the initial issue, this can also happen the container is broken.

Reply:
Yes you're right!

Reply:
Back to the main topic:
Where does this leave us? Do we want to catch two "groups" of exceptions:
- the first one allows for a fallback to the "default" option
- the second one requires at least an additional "ok" click from the user

Or do you prefer another solution?

Reply:
In my opinion, there's nothing useful the user could do at this point of time. Therefore, I would merge this PR as it is now.
What we could think about is: Do we need a detection if a container is broken? Is that check even possible?

Reply:
I think we could do at least a really basic check where we check that all files required by the engine and phoenicis are available. Don't we already have another issue about this?

Reply:
I'm not sure. I will check and open a new issue if not.